superset/connectors/base/models.py

```diff
@@ -396,7 +397,11 @@ def handle_single_value(value: Optional[FilterValue]) -> Optional[FilterValue]:
             ):
                 return datetime.utcfromtimestamp(value / 1000)
             if isinstance(value, str):
-                value = value.strip("\t\n'\"")
+                value = value.strip("\t\n")
+
+                quotes_value = re.findall(r"['|\"](.*?)['|\"]", value)
+                if len(quotes_value) == 1 and quotes_value[0] == value.strip("'\""):
+                    value = value.strip("'\"")
```

Comment:
Do we have any context on **why** we're stripping these quotes?

If the user typed quotes in the filter they should be passed as-is, and possibly escaped by the dialect if needed.

I think we should remove the quote stripping here, and if it breaks other things we should fix those things instead.

Reply:
Agree. It was added in https://github.com/apache/superset/pull/4651.

Reply:
Ahhh, that makes sense. From that PR it seems like the intent was allowing the user to represent `NULL` and an empty string by typing <kbd>NULL</kbd> and <kbd>''</kbd> respectively, so that they didn't get interpreted as the strings `"NULL"` and `"''"`.

Since if we have the option of writing custom SQL in filters, I would remove this logic.

Reply:
Hi Beto, I have added some unit test for this change. please review again. Thanks.